login/service_whitebox_test.go

```diff
@@ -66,10 +67,23 @@ func TestValidOAuthAccessToken(t *testing.T) {
 		AccessToken: configuration.GetGithubAuthToken(),
 		TokenType:   "Bearer",
 	}
-
 	emails, err := loginService.getUserEmails(context.Background(), accessToken)
-	assert.Nil(t, err)
-	assert.NotEmpty(t, emails)
+	var trials int // Number of tries to reach GitHub
+	if err != nil {
+		for trials = 0; trials < 10; trials++ {
+			emails, err = loginService.getUserEmails(context.Background(), accessToken)
+			if err == nil {
+				assert.NotEmpty(t, emails)
+				break
+			}
+			time.Sleep(5 * time.Second) // Pause before the next retry
+		}
+		if trials == 10 {
+			t.Error("Test failed, Maximum Retry limit reached", err) // Test failed after trial for 10 times
```

Comment:
@bartoszmajsak In the last attempt, trial will be 9, if there is a failure. trial will be 10 and it will be notified by t.Error and
if there is not a failure loop will break with trial = 9 and there won't be any notification of max retries reached.
If i am getting your point right, i think this will work.